cdap-watchdog/src/main/java/co/cask/cdap/logging/context/LoggingContextHelper.java

```diff
@@ -73,6 +76,18 @@ public static String getNamespacedBaseDir(NamespacedLocationFactory namespacedLo
     return namespacedLocationFactory.get(namespaceId.toId()).append(logBaseDir).toString();
   }
 
+  public static Location getNamespacedBaseDirLocation(final NamespacedLocationFactory namespacedLocationFactory,
```

Comment:
how come impersonation is not needed in the previous method? It seems like it should just call this method and then do a ``toString()``?